### PR TITLE
Add GET /internal/files. 

### PR DESCRIPTION
**Pull request by robinjhuang**
Retrieves list of files in models, output, and user directories.
To be used in model downloader / manager. 

Output is this format: 
`GET internal/files?directory=models`

```
{
    "files": [
     
        {
            "name": "gligen",
            "path": "gligen",
            "type": "directory"
        },
        {
            "name": "controlnet",
            "path": "controlnet",
            "type": "directory"
        },
        {
            "name": "t5_base.safetensors",
            "path": "clip/t5_base.safetensors",
            "type": "file",
            "size": 891646390
        },
        ....
    ]
}

```

### CODE REVIEW

**Review by huchenlei — APPROVED** [review]
LGTM!

File: api_server/services/file_service.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+class FileService:
+    def __init__(self, allowed_directories: Dict[str, str], file_system_ops: Optional[FileSystemOperations] = None):
+        self.allowed_directories: Dict[str, str] = allowed_directories
+        self.file_system_ops: FileSystemOperations = file_system_ops or FileSystemOperations()
+
+    def list_files(self, directory_key: str) -> List[FileSystemItem]:
+        if directory_key not in self.allowed_directories:
+            raise ValueError("Invalid directory key")
+        directory_path: str = self.allowed_directories[directory_key]
+        return self.file_system_ops.walk_directory(directory_path)
```
Comment (mcmonkey4eva) [inline api_server/services/file_service.py:13]: nl@eof

File: api_server/utils/file_operations.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        file_list: List[FileSystemItem] = []
+        for root, dirs, files in os.walk(directory):
+            for name in files:
+                file_path = os.path.join(root, name)
+                relative_path = os.path.relpath(file_path, directory)
+                file_list.append({
+                    "name": name,
+                    "path": relative_path,
+                    "type": "file",
+                    "size": os.path.getsize(file_path)
```
Comment (mcmonkey4eva) [inline api_server/utils/file_operations.py:32]: Maybe out of scope for this PR, but this function is going to be very slow without heavy caching of some form on any decently sized folder

**Comment by Charlweed** [comment]
Currently, this feature does not list the model directories gfpgan nor karlo, and it lists "photomaker" as a directory, though it is not in the filesystem.

**Pull request merged**